tests/tree_util_test.py

```diff
@@ -1501,6 +1501,14 @@ def test_tree_reduce_is_leaf(self):
       tree_util.tree_reduce(func, obj, is_leaf=is_leaf),
     )
 
+  def test_tree_reduce_associative(self):
```

Comment:
Since `jax.tree_util.tree_reduce_associative` is a public alias, we'll also need a test of that API (probably under `TreeTest` above). We should also have test coverage for the `is_leaf` option.